feat: update docs for profiling v2 (#13197)
getsentry/sentry-docs: docs/platforms/apple/common/profiling/index.mdx

Reviewer: Line 3: description: "Learn how to enable profiling in your app if it is not already set up."

Do we need the "if it is not already set up" part?

Author: No I don't think so, thanks!

[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
    options.dsn = @"___PUBLIC_DSN___";
    options.configureProfiling = ^(SentryProfileOptions *profiling) {
        profiling.sessionSampleRate = 1.f;
    };
}];
```

By default, `sessionSampleRate` is `0`, so you'll need to set it to a higher value to receive profile data. `sessionSampleRate` is evaluated once per user session and applies to any attempt to start a profile until the next user session starts. See <PlatformLink to="/configuration/releases/#sessions">user session documentation</PlatformLink> for more information on user sessions. 

See the subsections below to learn about the various ways the profiler can be started and stopped.

### Manual Lifecycle

> By default, the profiler can only be started and stopped manually with calls to `SentrySDK.startProfiler` and `SentrySDK.stopProfiler`. All code that executes on all threads in between those calls will be recorded. The configurations shown above demonstrate configuring manual profiling mode.

Reviewer: ```suggestion
By default, the profiler can only be started and stopped manually with calls to `SentrySDK.startProfiler` and `SentrySDK.stopProfiler`. All code that executes on all threads in between those calls will be recorded. The method shown above demonstrates configuring manual profiling mode.
```

> For example, if you wanted to profile everything that happens after starting a network request, and then updating a table view with the contents of the response, you could do is as follows (assuming you've already started the Sentry SDK with the options as shown above):

Reviewer: ```suggestion
For example, if you wanted to profile everything that happens after starting a network request, and then update a table view with the contents of the response, you could do it like this (assuming you've already started the Sentry SDK with the options as shown above):
```

[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
    options.dsn = @"___PUBLIC_DSN___";
    options.tracesSampleRate = @1.f;
    options.configureProfiling = ^(SentryProfileOptions *profiling) {
        profiling.sessionSampleRate = 1.f;
        profiling.lifecycle = SentryProfileLifecycleTrace;
    };
}];
```

<Alert>

> The `sessionSampleRate` for profiles is undersampled with respect to the `tracesSampleRate`.

Reviewer: I wonder if this could be worded more clearly? I can't tell if this is warning against using the `sessionSampleRate` for profiles when using it with `tracesSampleRate`? Or something else?

</Alert>

Check out the <PlatformLink to="/tracing/">tracing setup documentation</PlatformLink> for more detailed information on how to configure sampling for Sentry Tracing.

### App Starts

If configured with manual lifecycle, then a profile starts on the next app launch, and continues until you call `SentrySDK.stopProfiler`. 

> If configured with trace lifecycle, then app start profiles are attached to a special performance transaction operation called `app.launch` and displayed in the product simply as `launch`, which is stopped either when `SentrySDK.startWithOptions` is called, or, if <PlatformLink to="/tracing/instrumentation/automatic-instrumentation/#time-to-initial-display">TTID</PlatformLink>/<PlatformLink to="/tracing/instrumentation/automatic-instrumentation/#time-to-full-display">TTFD</PlatformLink> tracking is enabled, when the SDK determines that TTID/TTFD has been reached.

Reviewer: ```suggestion
If configured with trace lifecycle, app start profiles are attached to a special performance transaction operation called `app.launch` and displayed in the product as `launch`. It is stopped either when `SentrySDK.startWithOptions` is called, or, if <PlatformLink to="/tracing/instrumentation/automatic-instrumentation/#time-to-initial-display">TTID</PlatformLink>/<PlatformLink to="/tracing/instrumentation/automatic-instrumentation/#time-to-full-display">TTFD</PlatformLink> tracking is enabled, when the SDK determines that TTID/TTFD has been reached.
```
Also, is TTID/TTFD common enough where it is okay to leave as an acronym and not spell it out? I don't see the full name of either of these mentioned anywhere else in this article.

Author: I'll expand them on their first usages with parentheticals.

Every time `SentrySDK.startWithOptions` is called with app start profiling configured, a separate sample decision is generated with `sessionSampleRate` and stored until the next app launch (as well as `tracesSampleRate` if trace profile lifecycle is configured). The same sample decision will apply for the remainder of the profile session following that subsequent launch.

## Transaction Profiling (deprecated)

<Alert>

> Profiling configuration as explained below was originally introduced in SDK version `8.12.0` and is deprecated in favor of profiling V2 introduced in 8.49.0, explained at the beginning of this section.

Reviewer: ```suggestion
Profiling configuration, explained below, was originally introduced in SDK version `8.12.0`. It is now deprecated in favor of profiling V2 introduced in 8.49.0.
```

</Alert>

<Alert>

> Profiling depends on Sentry’s Tracing product being enabled beforehand. Check out the <PlatformLink to="/tracing/">tracing setup documentation</PlatformLink> for more detailed information on how to configure sampling.

Reviewer: ```suggestion
Profiling depends on Sentry’s Tracing product being enabled beforehand. Check out the <PlatformLink to="/tracing/">tracing setup documentation</PlatformLink> for detailed information on how to configure sampling.
```

</Alert>

Configure the sampling rates for traces and profiles to ensure they are nonzero so that some are recorded. The <PlatformIdentifier name="profiles-sample-rate" /> setting is _relative_ to the <PlatformIdentifier name="traces-sample-rate" /> setting.

```swift {tabTitle:Swift}
import Sentry

SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.tracesSampleRate = 1.0 // tracing must be enabled for profiling
    options.profilesSampleRate = 1.0 // see also `profilesSampler` if you need custom sampling logic
}
```

```objc {tabTitle:Objective-C}
@import Sentry;

[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
    options.dsn = @"___PUBLIC_DSN___";
    options.tracesSampleRate = @1.0; // tracing must be enabled for profiling
    options.profilesSampleRate = @1.0; // see also `profilesSampler` if you need custom sampling logic
}];
```

## Launch Profiling (deprecated)

<Alert>

> Launch profiling configuration as explained below was originally introduced in SDK version 8.21.0 and is deprecated in favor of profiling V2 configuration released in 8.49.0, explained at the beginning of this section.

Reviewer: ```suggestion
Launch profiling configuration, explained below, was originally introduced in SDK version 8.21.0. It is now deprecated in favor of profiling V2 configuration, released in 8.49.0.
```

Currently, launch profiles are attached to a special performance transaction operation called `app.launch` and displayed in the product simply as `launch`.

```swift {tabTitle:Swift}
import Sentry

SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.tracesSampleRate = 1.0 // tracing must be enabled for profiling
    options.profilesSampleRate = 1.0 // see also `profilesSampler` if you need custom sampling logic
    options.enableAppLaunchProfiling = true
}
```

```objc {tabTitle:Objective-C}
@import Sentry;

[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
    options.dsn = @"___PUBLIC_DSN___";
    options.tracesSampleRate = @1.0; // tracing must be enabled for profiling
    options.profilesSampleRate = @1.0; // see also `profilesSampler` if you need custom sampling logic
    options.enableAppLaunchProfiling = YES;
}];
```

## Continuous Profiling (deprecated)

<Alert>

> Continuous profiling as explained below was originally introduced in SDK version 8.36.0 and is deprecated in favor of profiling V2 configuration released in 8.49.0, explained at the beginning of this section.

Reviewer: ```suggestion
Continuous profiling, explained below, was originally introduced in SDK version 8.36.0. It is now deprecated in favor of profiling V2 configuration, released in 8.49.0.
```

Currently, launch profiles are attached to a special performance transaction operation called `app.launch` and displayed in the product simply as `launch`.

```swift {tabTitle:Swift}
import Sentry

SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.tracesSampleRate = 1.0 // tracing must be enabled for profiling
    options.profilesSampleRate = 1.0 // see also `profilesSampler` if you need custom sampling logic
    options.enableAppLaunchProfiling = true
}
```

```objc {tabTitle:Objective-C}
@import Sentry;

[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
    options.dsn = @"___PUBLIC_DSN___";
    options.tracesSampleRate = @1.0; // tracing must be enabled for profiling
    options.profilesSampleRate = @1.0; // see also `profilesSampler` if you need custom sampling logic
    options.enableAppLaunchProfiling = YES;
}];
```

## Continuous Profiling (deprecated)

<Alert>

> Continuous profiling as explained below was originally introduced in SDK version 8.36.0 and is deprecated in favor of profiling V2 configuration released in 8.49.0, explained at the beginning of this section.

Reviewer 2: Same here, call it Continuous Profiling Beta
